incubator/hnc/pkg/controllers/object_controller.go

```diff
@@ -284,14 +285,25 @@ func (r *ObjectReconciler) onDelete(ctx context.Context, log logr.Logger, nnm ty
 	return nil
 }
 
-func copyObject(inst *unstructured.Unstructured) *unstructured.Unstructured {
+func (r *ObjectReconciler) copyObject(nsNm string, inst *unstructured.Unstructured) *unstructured.Unstructured {
+	r.Forest.Lock()
+	defer r.Forest.Unlock()
+	ns := r.Forest.Get(nsNm)
+	if !ns.Exists() {
+		return nil
```

Comment:
If you look at the callers for copyObject, you'll see that a lot of them use the returned value immediately after it comes out of this function, so returning nil will result in a segfault. So for this solution to work, you'd have to modify all the call sites. That's a good "code smell" that this functionality doesn't belong here :)

Reply:
My nose is training!  Hahaha.

I should have listened to my gut when I started by adding the receiver.

Live and learn!